homeassistant/components/light/mqtt.py

```diff
@@ -192,6 +192,10 @@ def turn_on(self, **kwargs):
         """Turn the device on."""
         should_update = False
 
+        if not self._state:
```

Comment:
You are having the exact same problem as #1864 . I added a comment there that also applied to this PR:

> you will need to add support for an MQTT light that does not have a state topic, just a brightness topic. This should work as expected for `turn_on`, `turn_off` and `brightness_received`.
>
> This should also be accompanied by tests.